source/common/network/listen_socket_impl.cc

```diff
@@ -24,18 +24,22 @@ Api::SysCallIntResult ListenSocketImpl::bind(Network::Address::InstanceConstShar
   const Api::SysCallIntResult result = SocketImpl::bind(connection_info_provider_->localAddress());
   if (SOCKET_FAILURE(result.return_value_)) {
     close();
-    throw SocketBindException(fmt::format("cannot bind '{}': {}",
-                                          connection_info_provider_->localAddress()->asString(),
-                                          errorDetails(result.errno_)),
-                              result.errno_);
+    const std::string error =
+        fmt::format("cannot bind '{}': {}", connection_info_provider_->localAddress()->asString(),
+                    errorDetails(result.errno_));
+#ifdef ENVOY_DISABLE_EXCEPTIONS
+    PANIC(error);
+#else
+    throw SocketBindException(error, result.errno_);
```

Comment:
This can't use throwExceptionOrPanic() because this exception type's constructor is special?

Reply:
yeah, I tried to normalize the constructor like I did elsewhere but it turns out we use the errno in tests to keep listening on random ports until we get one that works so it's non-trivial